### DISCUSSION TITLE
Offre zen Flex

### DISCUSSION THREAD

**Discussion by laurentftech** [MyElectricalData/myelectricaldata_import#468] [Ideas]
Bonjour. 
Est-ce qu'il serait envisageable de proposer le monitoring de l'offre zen flex ? C'est le même genre que Tempo mais les jours "sobriété" ne sont pas les mêmes et les HP/HC sont spécifiques.
Les jours "sobriété" sont disponibles avec une API EDF ici:
https://particulier.edf.fr/services/rest/opm/getOPMStatut?dateRelevant=2024-01-19

**Comment by m4dm4rtig4n** [comment]
J'y ai pensais hier justement, mais effectivement on ne sais pas récupérer les jours de sobriété actuellement.
Je pensais partir sur les jours Rouge Tempo et retirer 2j car je pense que les 20j de sobriété seront forcement pendant les jours rouge Tempo.

**Reply by laurentftech** [reply]
C'est ce que je croyais mais en fait non si j'en crois le calendrier trouvé ici et rempli à la main par un utilisateur. Il serait intéressant de savoir si quelque chose de plus industriel est préparé par EDF. 

https://masfaraud.github.io/ZenFlex/

**Reply by m4dm4rtig4n** [reply]
Effectivement :/

**Reply by laurentftech** [reply]
Pour info, j'ai contacté RTE. Réponse ce matin:
L’option sobriété est un service propre à EDF.
A ce jour, RTE ne propose pas d’API dans ce sens et à notre connaissance cela n’est pas prévu.
 
C'est bien à EDF de jouer.

**Reply by laurentftech** [reply]
Bonjour. 
Il y a bien une API EDF pour récupérer les jours sobriété 😀 

https://particulier.edf.fr/services/rest/opm/getOPMStatut?dateRelevant=2024-01-19

**Reply by neofutur** [reply]
J utilisais cette URL pour mon home assistant, mais cette API ( https://particulier.edf.fr/services/rest/opm/getOPMStatut?dateRelevant=2026-09-19 ) ne semble plus exister.  Savez vous si ils ont changé l URL de l API ?

**Comment by laurentftech** [comment]
Bonjour. 

Est-ce qu'il serait possible d'intégrer les offres Flex Weekend dans MyElectricalData ? En fait, ce serait déjà vraiment top d'avoir les jours Sobriété, Normal et Bonus au moins dans le Dashboard Energy de Homeassistant. J'ai essayé quelques manip pour le faire à posteriori dans HA mais c'est trop compliqué (pour moi en tout cas, je n'arrive pas à le faire avec les décomptes horaires). Avec quelques tips, je pourrais peut-être y arriver (en fait je n'arrive pas à savoir comment manipuler les données horaires dans HA, c'est possible ou pas?). 
Les jours sont disponibles sur l'API Rest EDF avec le mapping:
- 'RAS' : 'Normal'
- 'ZENF_PM' : 'Sobriété'
- 'ZENF_BONIF' : 'Bonus'
- 'NON_DETERMINE' : 'Non déterminé'

Pour plus de simplicité les coûts en EUR/kWh pourraient être gérés librement dans HA mais avoir déjà les mesures en HP/HC pour chacune des catégories de jour serait énorme. 

Edit : à la réflexion, je vais essayer d'obtenir cela par un croisement entre la base Influxdb depuis Homeassistant et une nouvelle base Jours Flex
